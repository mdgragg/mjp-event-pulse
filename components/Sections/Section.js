import styled from 'styled-components';
import React from 'react';
import { Grid, Button, Typography } from '@material-ui/core';

const StyledSection = styled.div`
  padding: 2em;
  min-height: 80vh;
  width: 100%;
  min-height: ${(props) => props.minHeight || 'auto'};
  background-color: ${(props) => props.bgColor || 'white'};
  margin-left: auto;
  margin-right: auto;
  @media (max-width: 768px) {
    padding: 1rem;
  }
`;

const Section = (props) => {
  return <StyledSection {...props}>{props.children}</StyledSection>;
};

export default Section;
