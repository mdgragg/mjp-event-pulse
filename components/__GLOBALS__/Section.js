import  styled  from 'styled-components';
import React from 'react';
import { Grid, Button } from '@material-ui/core'

const StyledSection = styled.div`
padding: 2em;
width: 100%;
min-height: 60vh;
margin-left: auto;
margin-right: auto;
@media (max-width: 768px){
    padding: 0;
}
:nth-child(even){
    background-color: #F2f2f2;
}
h1{
    margin-top: 20px;
    margin-bottom: 50px;
}
Button.seeAll{
    margin-top: 50px;
    margin-bottom: 50px;
    background-color: ${props => props.theme.secondary};
    font-weight: bold;
    color: white;
    :hover{
        background-color: ${props => props.theme.hover};

    }
}
`;

const InnerContent = styled.div` 
max-width: 1400px;
margin-left: auto;
margin-right: auto;
` 


const Section = (props) => {
    return (
        <StyledSection>
            <InnerContent>
            <center> <h1>{props.title} </h1></center>
                {props.children}
                {props.showButton ? <center> <Button className="seeAll">See All {props.title}</Button></center> : "" }
             
              

                </InnerContent>
        </StyledSection>
    );
};

export default Section;