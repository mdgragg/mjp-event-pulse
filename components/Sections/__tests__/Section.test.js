/**
 * @jest-environment jsdom
 */

import Section from '../Section';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Section', () => {
  it('Should render', () => {
    const counter = renderer.create(<Section />).toJSON();
    expect(counter).toMatchSnapshot();
  });
});
