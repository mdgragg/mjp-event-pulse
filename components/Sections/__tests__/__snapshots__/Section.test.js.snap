// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counters Should render 1`] = `
<div
  className="Section__StyledSection-jAnJfD jGjdV"
/>
`;

exports[`Section Should render 1`] = `
<div
  className="Section__StyledSection-jAnJfD jGjdV"
/>
`;
